api/asymmetric_signin.go

```diff
@@ -65,6 +71,15 @@ func (a *API) GetChallengeToken(w http.ResponseWriter, r *http.Request) error {
 				Provider: "AsymmetricKey",
 				Aud:      aud,
 			})
+
+			if terr != nil {
+				return terr
+			}
+
+			if config.FirstUserSuperAdmin && first {
+				terr = user.SetSuperAdmin(tx)
+			}
```

Comment:
I would move this logic into (a *API) signupNewUser


```
	err = conn.Transaction(func(tx *storage.Connection) error {
		var terr error
		if terr = tx.Create(user); terr != nil {
			return internalServerError("Database error saving new user").WithInternalError(terr)
		}
		if config.FirstUserSuperAdmin && first {
				terr = user.SetRole(tx, "superadmin")
			} else {
			terr = user.SetRole(tx, config.JWT.DefaultGroupName); terr != nil {
			return internalServerError("Database error updating user").WithInternalError(terr)
		}
			}
	
		if terr = triggerEventHooks(ctx, tx, ValidateEvent, user, instanceID, config); terr != nil {
			return terr
		}
		return nil
	})
	if err != nil {
		return nil, err
	}

	return user, nil
```

Reply:
smth like this

Also models.IsFirstUser is a bit misleading here, as at this moment we don't have user.
I would rename to models.AnyUser or smth like this.